### PR TITLE
feat: use interact and introspect separately to communicate with idx endpoints

### PR DESCRIPTION
**Pull request by shuowu**
interact: call idx.interact and manage meta in storage
introspect: call idx.introspect and manage intermediate idx response in storage (not implemented yet)

### CODE REVIEW

File: test/support/jest/jest.config.js [hunk 1]
```diff
@@ -26,7 +26,8 @@ module.exports = {
     '<rootDir>/test/support/jest/jest.setup.js'
   ],
   'testMatch': [
-    '**/test/spec/**/*.{js,ts}'
+    // '**/test/spec/**/*.{js,ts}'
+    '**/test/spec/idx/recoverPassword.{js,ts}'
```
Comment (aarongranick-okta) [inline test/support/jest/jest.config.js:30]: make sure to change this back

**Review by aarongranick-okta — CHANGES_REQUESTED** [review]
looks good, just remove the filter on jest config and ensure other tests are passing

File: test/support/jest/jest.config.js (hunk 1 above)
Comment (shuowu) [reply test/support/jest/jest.config.js:30]: for sure, did git add all... I should stash those unfinished changes ...

**Comment by shuowu** [comment]
@aarongranick-okta ready for review, unit test for `introspect` will be added in the next pr with storage implementation.

File: test/support/jest/jest.config.js (hunk 1 above)
Comment (shuowu) [reply test/support/jest/jest.config.js]: removed filter

**Review by aarongranick-okta — APPROVED** [review]

**Pull request closed without merging**